backend/src/main/kotlin/app/ehrenamtskarte/backend/freinet/webservice/FreinetApplicationMutationService.kt

```diff
@@ -0,0 +1,145 @@
+import io.ktor.http.HttpMethod
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.URLProtocol
+import io.ktor.http.path
+import kotlinx.coroutines.runBlocking
+import kotlinx.serialization.json.Json
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.slf4j.LoggerFactory
+
+@Suppress("unused")
+class FreinetApplicationMutationService {
+    private val logger = LoggerFactory.getLogger(FreinetApplicationMutationService::class.java)
+    private val httpClient = HttpClient {
+        install(HttpRequestRetry) {
+            retryOnServerErrors(maxRetries = 3)
+            retryOnException(maxRetries = 3, retryOnTimeout = false)
+            exponentialDelay()
+        }
+        HttpResponseValidator {
+            validateResponse { response ->
+                if (response.status != HttpStatusCode.OK) {
+                    throw Exception(response.status.toString())
+                }
+            }
+        }
+    }
+
+    private val jsonParser = Json { ignoreUnknownKeys = true }
+
+    @GraphQLDescription("Send application info to Freinet")
+    fun sendApplicationDataToFreinet(applicationId: Int, project: String, dfe: DataFetchingEnvironment): Boolean {
+        val context = dfe.graphQlContext.context
+        val admin = context.getAdministrator()
+        val projectConfig = dfe.graphQlContext.context.backendConfiguration.getProjectConfig(
+            project,
+        )
+        if (projectConfig.freinet == null) {
+            throw NotImplementedException()
+        }
+
+        return transaction {
+            val application = ApplicationRepository.findByIds(listOf(applicationId)).firstOrNull()
+                ?: throw NotFoundException("Application not found")
+
+            val regionId = application.regionId.value
+
+            if (!Authorizer.mayViewApplicationsInRegion(admin, regionId)) {
+                logger.warn("unauthorized access")
+                throw UnauthorizedException()
+            }
+
+            val freinetAgency = FreinetAgencyRepository.getFreinetAgencyByRegionId(regionId)
+                ?: throw FreinetAgencyNotFoundException()
+
+            if (!freinetAgency.dataTransferActivated) {
+                // Freinet is disabled for this region
+                throw FreinetDataTransferNotActivatedException()
+            }
+
+            val applicationData = application.jsonValue
+            val objectMapper = jacksonObjectMapper()
+            val jsonNode = objectMapper.readTree(applicationData)
+
+            val personalDataNode = jsonNode
+                .path("value").firstOrNull { it["name"].asText() == "personalData" }
+
+            fun JsonNode.findValueByName(fieldName: String): String? =
+                this.firstOrNull { it["name"].asText() == fieldName }
+                    ?.get("value")
+                    ?.asText()
+
+            val firstName = personalDataNode?.get("value")?.findValueByName("forenames").orEmpty()
```

Comment:
🔧 Please move the logic of external api calls in a separate class as i requested already before